app/screens/draft_options/send_draft/index.tsx

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase, withObservables} from '@nozbe/watermelondb/react';
+import {combineLatest, of as of$} from 'rxjs';
+import {switchMap} from 'rxjs/operators';
+
+import {INITIAL_PRIORITY} from '@components/post_draft/send_handler/send_handler';
+import {General, Permissions} from '@constants';
+import {MAX_MESSAGE_LENGTH_FALLBACK} from '@constants/post_draft';
+import {observeChannel, observeChannelInfo} from '@queries/servers/channel';
+import {queryAllCustomEmojis} from '@queries/servers/custom_emoji';
+import {observeFirstDraft, queryDraft} from '@queries/servers/drafts';
+import {observePermissionForChannel} from '@queries/servers/role';
+import {observeConfigBooleanValue, observeConfigIntValue, observeCurrentUserId} from '@queries/servers/system';
+import {observeUser} from '@queries/servers/user';
+
+import SendDraft from './send_draft';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+
+type Props ={
+    rootId: string;
+    channelId: string;
+} & WithDatabaseArgs;
+
+const enhanced = withObservables([], ({database, rootId, channelId}: Props) => {
```

Comment:
0/5 I wonder if this can be shared with the post input component.

That being said... probably they are different enough to better keep them separate.

Reply:
Yup, it is better to keep them separate because of their tightly bound nature.

Reply:
What makes it tightly bound?